include/deal.II/matrix_free/dof_info.h

```diff
@@ -432,7 +432,7 @@ namespace internal
       AssertIndexRange (row, row_starts.size()-1);
       const unsigned int index = row_starts[row][0];
       AssertIndexRange(index, dof_indices.size()+1);
-      return dof_indices.empty() ? nullptr : &dof_indices[0] + index;
+      return dof_indices.empty() ? nullptr : dof_indices.data() + index;
```

Comment:
why not ``dof_indices[index]``?

Reply:
The return type is a pointer, not a value (or reference, or iterator). I think that this way is a little better than writing `&dof_indices[index]`, but both would work.

Reply:
It depends. @drwells 's notation implicitly assumes that the elements of the array are contiguous in memory. @tjhei's notation does not make this assumption.

Reply:
All users of this function make this assumption anyway, so there is no big difference either way. I do not have strong opinion - I used to tend to @tjhei's suggestion some years ago but now I tend towards @drwells's approach, but it's really only a matter of taste.

Reply:
I agree, it's a matter mostly of taste. I'm happy to be deferential to the person who put in the time to write this patch.

Reply:
I don't have a strong opinion either, but I think ``&dof_indices[index]`` makes it more obvious that this points to the index-th element. I had to look up the type of ``dof_indices`` because I was unsure what was going on here.

Reply:
Good point. I'll change this.